machine/v1/types_controlplanemachineset.go

```diff
@@ -248,6 +249,10 @@ type FailureDomains struct {
 	// GCP configures failure domain information for the GCP platform.
 	// +optional
 	GCP *[]GCPFailureDomain `json:"gcp,omitempty"`
+
+	// OpenStack configures failure domain information for the OpenStack platform.
+	// +optional
+	OpenStack []OpenStackFailureDomain `json:"openstack,omitempty"`
```

Comment:
@JoelSpeed I've updated it here to not be a pointer, and updated CCPMSO as well.

Reply:
I guess I should add a comment here explaining why we don't use a pointer.